File: submissions/avik_nilam/problem4.cpp
#include<stdio.h>
int prime(int);
int magic(int);
int main()
{
	int x,y,a,b,i,j;
	scanf("%d%d",&x,&y);
	if(x>y)
	{
		printf("Invalid Input");
	}
	else
	{
	for(i=x;i<=y;i++)
	{
		a=prime(i);
		b=magic(i);
		//printf("%d\n",i);
		if(a==0&&b==1)
		{
			printf("%d ",i);
		}
		else
		continue;
	}
	}
	return 0;
	
}
int magic(int n)
{
	if(n==1)
	{
		return 1;
	}
	else if(n/10==0)
	{
		return 0;
	}
	else
	{
		int a,size,i;
		i=n;
		a=0;
		size=0;
		while(i!=0)
		{
			i=i/10;
			size++;
		}
	//	printf("%d\n",size);
		for(i=0;i<size;i++)
		{
			a=a+n%10;
			n=n/10;
		}
	//	printf("%d",a);
		magic(a);
	}
}
int prime(int n)
{
	int i,x;
	for(i=2;i<=n-1;i++)
	{
		if(n%i==0)
		{
			x=0;
			break;
		}
		else
		{
			x=1;
		}
	}
	return x;
}

File: submissions/sagar_sandeep/problem4.java
import java.util.*;
class problem4
{
    public static int magic(int n)
    {
        while(nod(n)>1)
        {
            n=sod(n);
        }
        if(n==1)
        return 1;
        else 
        return 0;
    }
    public static int nod(int n)
    {
        int c=0;
        while(n>0)
        {
            n/=10;
            c++;
        }
        return c;
    }
    public static int sod(int n)
    {
        int s=0;
        while(n>0)
        {
            s+=(n%10);
            n/=10;
        }
        return s;
    }
    public static int iscom(int n)
    {
        int i=0,c=0;
        for(i=2;i<n;i++)
        {
            if(n%i==0)
            c++;
        }
        if(c!=0)
        return 1;
        else 
        return 0;
    }    
    public static void main(String[] args)
    {
        Scanner sc=new Scanner(System.in);
        int m=sc.nextInt(),n=sc.nextInt(),i;
        if(m<n)
        {
        for(i=m;i<=n;i++)
        {
            if(iscom(i)==1&&magic(i)==1)
            System.out.print(i+"  ");
        }
    }
    else
     System.out.print("invalid Input");
    }
}
            